### ISSUE TITLE
[New Sensor]: Xiaomi Human Presence Sensor

### ISSUE THREAD

**Issue by merdok** [custom-components/ble_monitor#1377]
### Sensor Description

Xiaomi Human Presence Sensor

### Additional information

https://aliexpress.com/item/1005007045530291.html
Product number: XMOSB01XS
Wireless connections: Bluetooth 5.0
Battery Type: CR2450
Net Weight: 47.6g
Detection distance: 
Human movement detection up to 6 meters
Human presence detection up to 4 meters
detection angle: about 130°
Operating temperature: -10C° ~ 45C°
Working humidity: 0-95%RH (non-condensing)

**token: b296b456e3974e757a349ff4**

New xiaomi human presence sensor.

Could it also be added to the Xiaomi BLE integration?

### BLE advertisements

```shell

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:43:57
Event Type: NonConnectable Undirected
Payload:
01-06-16-95-FE-48-59-83-46-89-44-FO-CB-3D-36-1A-84-04-00-00-1D-F7-3C-C3

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:44:54
Event Type: NonConnectableUndirected
Payload:
01-06-16-95-FE-58-59-83-46-8A-97-95-B8-67-48-D4-32-OF-1F-99-04-00-00-7B-8D-BF-6C

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:45:00
Event Type: NonConnectable Undirected
Payload:
01-06-16-95-FE-58-59-83-46-8B-97-95-B8-67-48-D4-C3-7C-0E-EE-04-00-00-C6-E9-26-D8

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:45:17
Event Type: Non Connectable Undirected
Payload:
01-06-16-95-FE-10-59-83-46-8B-97-95-B8-67-48-D4

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:45:18
Event Type: NonConnectable Undirected
Payload: 01-06-16-95-FE-48-59-83-46-8C-70-ED-BE-15-7B-2E-14-04-00-00-AF-B3-D0-E1

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:45:56
Event Type: NonConnectable Undirected
Payload:
01-06-16-95-FE-48-59-83-46-8D-17-A3-75-FC-78-53-AD-04-00-00-F7-4E-F9-0D

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:46:00
Event Type: Non Connectable Undirected
Payload: 01-06-16-95-FE-48-59-83-46-8E-EE-01-1B-C7-E3-54-8F-04-00-00-EC-7A-26-64

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:42:14
Event Type: ConnectableUndirected
Payload: 01-06-16-95-FE-10-59-83-46-86-97-95-B8-67-48-D4

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:43:19
Event Type: NonConnectable Undirected
Payload:
01-06-16-95-FE-48-59-83-46-88-93-92-A4-60-09-AD-88-04-00-00-12-D9-E5-BB

Address: D44867B89597
Address Type: Public
Timestamp: 12/07/2024 23:42:16
Event Type: Non Connectable Undirected
Payload: 01-06-16-95-FE-48-59-83-46-87-69-AE-4E-45-48-82-B1-04-00-00-20-CO-E0-61

```

**Comment by yangqian** [comment]
According to the specs, it is a ble mesh device. So it may require active connection?
I don't why Xiaomi made it that way, seems unnecessary.

**Comment by merdok** [comment]
I mean it sends an advertisement when a person is detected, so not sure if it really needs an active connection... Also how would they even be able to achieve 3 years of battery life on a single cell battery with an active connection?

**Comment by yangqian** [comment]
> I mean it sends an advertisement when a person is detected, so not sure if it really needs an active connection... Also how would they even be able to achieve 3 years of battery life on a single cell battery with an active connection?

Yeah, seems strange, it may be putting all the information just in advertisement. But It might be a new protocal.

Does it have a BLE KEY rather than the token? It should be longer.

**Comment by merdok** [comment]
Do not see any ble key entry in the micloud device info.

**Comment by niaosuan** [comment]
Is this sensor can be added into HA?
From review It is an all-In-one sensor It solved presences and movement sensor weaknesses.

**Comment by ldfandian** [comment]
> Is this sensor can be added into HA? From review It is an all-In-one sensor It solved presences and movement sensor weaknesses.

[The pull request](https://github.com/home-assistant/core/pull/124751) to let "xiaomi_ble" integration supports it has just been merged.

**Comment by ldfandian** [comment]
> According to the specs, it is a ble mesh device. So it may require active connection? I don't why Xiaomi made it that way, seems unnecessary.

No, it is simply a BLE device, not a BLE mesh device.

**Comment by cybe** [comment]
Not yet in `2024.9.0`, for those who are wondering.

**Comment by ldfandian** [comment]
> Not yet in `2024.9.0`, for those who are wondering.

Yeah, it will probably be released in 2024.10.0~

If you are impatient, you can use:
- Nightly dev container: The easiest method if you’re comfortable with a dev build.
- Custom branch: If you’re worried about dev build stability, you can use my private branch ( https://github.com/ldfandian/homeassistant-core/tree/2024.8.3-xiaomi-ble-0.32.0 ).
![image](https://github.com/user-attachments/assets/d08b3cce-3d7c-4182-aad7-bee76837dbaf)

**Comment by lee-jinhwan** [comment]
@ldfandian 
I'm using the dev version, but the device is not being added, is there something else going on?
(I'm receiving events through the ESPHome BLE Gateway).

**Comment by lee-jinhwan** [comment]
`ble_monitor` also need to support `XMOSB01XS`.

Here is my gist file. 
https://gist.github.com/lee-jinhwan/487847b7f62868d85f0f3186a90127bc

**Comment by ldfandian** [comment]
> @ldfandian I'm using the dev(`2024.10.0.dev202409220231`) version, but the device is not being added, is there something else going on? (I'm receiving events through the ESPHome BLE Gateway).

I guess you already know the answer~

"Bluetooth Proxy" and "BLE Gateway" are different things.

[https://custom-components.github.io/ble_monitor/parse_data](https://custom-components.github.io/ble_monitor/parse_data)
![image](https://github.com/user-attachments/assets/a0b92511-2c4a-478f-83f4-1eab76afd065)

**Comment by ldfandian** [comment]
:tada: :tada: :tada: [update@2024-10-04] The change is already included in the 2024.10.0 stable release!

**Comment by LeoI5** [comment]
> `ble_monitor` also need to support `XMOSB01XS`.
> 
> Here is my gist file. https://gist.github.com/lee-jinhwan/487847b7f62868d85f0f3186a90127bc

It works. Thanks

**Comment by LeoI5** [comment]
> 🎉 🎉 🎉 [update@2024-10-04] The change is already included in the 2024.10.0 stable release!

Are you sure? Nothing is inside update@2024-10-04 about XMOSB01XS :(

**Comment by LeoI5** [comment]
Still wait for support XMOSB01XS

**Comment by andrewjswan** [comment]
Are there any plans to add this sensor to BLE Monitor?

**Comment by Ernst79** [comment]
Added in 13.8.0
